generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/PrunedConstraint.java

```diff
@@ -0,0 +1,61 @@
+package com.scottlogic.deg.generator.walker.reductive;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.constraints.atomic.AtomicConstraint;
+import com.scottlogic.deg.generator.decisiontree.ConstraintNode;
+import com.scottlogic.deg.generator.decisiontree.DecisionNode;
+import com.scottlogic.deg.generator.decisiontree.TreeConstraintNode;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpec;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class PrunedConstraint {
```

Comment:
I think this type could be made immutable with relative ease, which might make things a bit easier to comprehend?

Or does it *need* to be mutable for the process?